src/main/kotlin/org/rust/ide/inspections/RsUnresolvedReferenceInspection.kt

```diff
@@ -84,21 +86,29 @@ class RsUnresolvedReferenceInspection : RsLocalInspectionTool() {
 
         val referenceName = element.referenceName
         val description = if (referenceName == null) "Unresolved reference" else "Unresolved reference: `$referenceName`"
-        var fix: LocalQuickFix? = null
+        val fixes = mutableListOf<LocalQuickFix>()
         if (candidates != null && candidates.isNotEmpty()) {
-            fix = if (RsCodeInsightSettings.getInstance().showImportPopup) {
+            val importFix = if (RsCodeInsightSettings.getInstance().showImportPopup) {
                 AutoImportHintFix(element, context.type, candidates[0].info.usePath, candidates.size > 1)
             } else {
                 AutoImportFix(element, context.type)
             }
+            fixes.add(importFix)
+
+            if (element is RsPath &&
```

Comment:
I extracted the method, but I'm having trouble with tests. In the test `test add extern crate with 2015 edition`, `crateRoot` is null for the unresolved path, therefore `extern crate` is not added. How can I set up the test so that `crateRoot` is set properly?

Reply:
Or, it's simple. Just insert `extern crate item` **before** replacing old path with new one 😄